sns/parity_math.c

```diff
@@ -500,6 +651,35 @@ static void reed_solomon_encode(struct m0_parity_math *math,
 
 #undef PARITY_MATH_REGION_ENABLE
 }
+#endif /* __KERNEL__ */
+
+#ifndef __KERNEL__
+static void isal_encode(struct m0_parity_math *math,
+			const struct m0_buf *data,
+			struct m0_buf *parity)
+{
+	uint32_t   pi; /* parity unit index. */
+	uint32_t   di; /* data unit index. */
+	uint32_t   block_size = data[0].b_nob;
+
+	math->data_frags[0] = (uint8_t *)data[0].b_addr;
+	for (di = 1; di < math->pmi_data_count; ++di) {
+		M0_ASSERT(block_size == data[di].b_nob);
+		math->data_frags[di] = (uint8_t *)data[di].b_addr;
+	}
```

Comment:
Any reason to handle the "math->data_frags[0]" with exception?
Why not in the "for" loop and the "di" from 0?

Reply:
In loop we are comparing block_size, which is data[0].b_nob, with b_nob of every data. If we start with di = 0, we are again comparing block_size with data[0].b_nob. Hence we started with di = 1.